main.cpp

```diff
@@ -1 +1,411 @@
+/**
+ * @name TP_TicTacToe
+ *
+ * @author Guillaume Gonin
+ * @date 29.09.2022
+ */
+
 #include <iostream>
+#include <array>
+#include <string>
+
+//the game array set by user/s
+std::array<std::array<int, 3>, 3> GAME_ARR = {{{0, 0, 0}, {0, 0, 0}, {0, 0, 0}}};
+
+
+/**
+ * Check if the user given had win
+ * @param userToCheck the user value to check
+ * @return true if he win else false
+ */
+bool checkWin(int userToCheck) {
+    if (GAME_ARR[1][1] == userToCheck) {
+        if (((GAME_ARR[0][0] == userToCheck) && (GAME_ARR[2][2] == userToCheck)) ||
+            ((GAME_ARR[1][0] == userToCheck) && (GAME_ARR[1][2] == userToCheck)) ||
+            ((GAME_ARR[2][0] == userToCheck) && (GAME_ARR[0][2] == userToCheck)) ||
+            ((GAME_ARR[0][1] == userToCheck) && (GAME_ARR[2][1] == userToCheck)))
+            return true;
+        else return false;
+    } else if (GAME_ARR[0][0] == userToCheck) {
+        if (((GAME_ARR[0][1] == userToCheck) && (GAME_ARR[0][2] == userToCheck)) ||
+            ((GAME_ARR[1][0] == userToCheck) && (GAME_ARR[2][0] == userToCheck)))
+            return true;
+        else return false;
+    } else if (GAME_ARR[2][2] == userToCheck) {
+        if (((GAME_ARR[0][2] == userToCheck) && (GAME_ARR[1][2] == userToCheck)) ||
+            ((GAME_ARR[2][0] == userToCheck) && (GAME_ARR[2][1] == userToCheck)))
+            return true;
+        else return false;
+    } else return false;
+}
+
+/**
+ * Check if the user given had win
+ * :this method can navigate every size of array (searching for 3 case checked)
+ * @param userToCheck the user value to check
+ * @return true if he win else false
+ */
+bool checkWinR(int userToCheck) {
```

Comment:
This method is useless and too smart - always do the simplest way possible. Also, the simpler `checkWin` is not used.